subsys/net/lib/nrf_cloud/src/nrf_cloud_pgps.c

```diff
@@ -0,0 +1,1647 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <nrfx_nvmc.h>
+#include <device.h>
+#include <drivers/gps.h>
+#include <storage/stream_flash.h>
+#include <net/socket.h>
+#include <nrf_socket.h>
+
+#include <cJSON.h>
+#include <cJSON_os.h>
+#include <modem/modem_info.h>
+#include <date_time.h>
+#include <net/nrf_cloud_agps.h>
+#include <net/nrf_cloud_pgps.h>
+#include <settings/settings.h>
+#include <power/reboot.h>
+#include <logging/log_ctrl.h>
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(nrf_cloud_pgps, CONFIG_NRF_CLOUD_GPS_LOG_LEVEL);
+
+#include "nrf_cloud_transport.h"
+#include "nrf_cloud_pgps_schema_v1.h"
+#include "nrf_cloud_pgps_utils.h"
+
+#define FORCE_HTTP_DL			0 /* set to 1 to force HTTP instead of HTTPS */
+#define PGPS_DEBUG			0 /* set to 1 for extra logging */
+
+#define PREDICTION_PERIOD		CONFIG_NRF_CLOUD_PGPS_PREDICTION_PERIOD
+#define REPLACEMENT_THRESHOLD		CONFIG_NRF_CLOUD_PGPS_REPLACEMENT_THRESHOLD
+#define SEC_TAG				CONFIG_NRF_CLOUD_SEC_TAG
+#define FRAGMENT_SIZE			CONFIG_NRF_CLOUD_PGPS_DOWNLOAD_FRAGMENT_SIZE
+#define PREDICTION_MIDPOINT_SHIFT_SEC	(120 * SEC_PER_MIN)
```

Comment:
If CONFIG_NRF_CLOUD_PGPS_PREDICTION_PERIOD sets the ephemeris fit interval, then PREDICTION_MIDPOINT_SHIFT_SEC needs to be calculated based on that.

Reply:
No, CONFIG_NRF_CLOUD_PGPS_PREDICTION_PERIOD has no direct effect on the fit interval -- HERE only supports 4 hours for the fit interval.  HERE always sends our server 120 minute predictions, with the fit interval being 4 hours for each prediction.  HERE was unwilling to provide any other fit interval.

If CONFIG_NRF_CLOUD_PGPS_PREDICTION_PERIOD = 240 (minutes), then this value lines up the predictions well so the full fit interval is used.

But, when CONFIG_NRF_CLOUD_PGPS_PREDICTION_PERIOD = 120, the logic in `start_expiration_timer()` is slightly less than optimal -- it doesn't try to use the sweet spot in the middle of each prediction.